numpy/_globals.py

```diff
@@ -89,3 +92,23 @@ def __repr__(self):
 
 
 _NoValue = _NoValueType()
+
+
+class _CopyMode(enum.Enum):
+
+    ALWAYS = True
+    IF_NEEDED = False
+    NEVER = 2
+
+    def __bool__(self):
+        # For backwards compatiblity
+        if self == _CopyMode.ALWAYS:
+            return True
+
+        if self == _CopyMode.IF_NEEDED:
+            return False
+
+        raise TypeError(f"{self} is neither True nor False.")
```

Comment:
```suggestion
        raise ValueError(f"{self} is neither True nor False.")
```
Should this not be a `ValueError`? Considering the error is triggered via truth-testing an unsupported enumeration value.